pkgs/development/python-modules/cot/default.nix

```diff
@@ -0,0 +1,66 @@
+{ stdenv, buildPythonPackage, fetchPypi, pythonOlder, isPy3k
+, argcomplete, colorlog, pyvmomi, requests, verboselogs
+, psutil, pyopenssl, setuptools
+, mock, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "cot";
+  version = "2.2.1";
+
+  disabled = !isPy3k;
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "f4b3553415f90daac656f89d3e82e79b3d751793239bb173a683b4cc0ceb2635";
+  };
+
+  propagatedBuildInputs = [ colorlog pyvmomi requests verboselogs pyopenssl setuptools ]
+  ++ stdenv.lib.optional (pythonOlder "3.3") psutil;
+
+  checkInputs = [ mock pytest ];
+
+  checkPhase = ''
+    runHook preCheck
+
+    # ignore files containing failing tests
+    pytest \
+      --ignore=COT/commands/tests/test_add_disk.py \
```

Comment:
Try to use `disabledTests` instead.

Reply:
Ah, did not see that, though I did see some custom implementations (e.g. `skipTests`, etc)
I will see if this works.

Reply:
disabledTests only works with `pytestCheckHook`

Reply:
@jonringer Yeah, I noticed that at some point, and then found the hooks needed for the other libraries, but I wasn't able to determine what I needed to input to get `disabledTests` working properly.

Just using `pytestCheckHook` and `pytestFlagsArray` looks a bit nicer, anyhow.
I'm sure it would be even better if I got the `disabledTests` arguments right... 

<details>
<summary><span><code>pytestCheckHook</span></code> and <span><code>pytestFlagsArray</code></span></summary>

``` diff
diff --git a/pkgs/development/python-modules/cot/default.nix b/pkgs/development/python-modules/cot/default.nix
index f6baef515f4..4a5da3204a4 100644
--- a/pkgs/development/python-modules/cot/default.nix
+++ b/pkgs/development/python-modules/cot/default.nix
@@ -1,7 +1,7 @@
 { stdenv, buildPythonPackage, fetchPypi, pythonOlder, isPy3k
 , argcomplete, colorlog, pyvmomi, requests, verboselogs
 , psutil, pyopenssl, setuptools
-, mock, pytest
+, mock, pytest-mock, pytestCheckHook
 }:

 buildPythonPackage rec {
@@ -18,35 +18,28 @@ buildPythonPackage rec {
   propagatedBuildInputs = [ colorlog pyvmomi requests verboselogs pyopenssl setuptools ]
   ++ stdenv.lib.optional (pythonOlder "3.3") psutil;

-  checkInputs = [ mock pytest ];
-
-  checkPhase = ''
-    runHook preCheck
-
-    # ignore files containing failing tests
-    pytest \
-      --ignore=COT/commands/tests/test_add_disk.py \
-      --ignore=COT/commands/tests/test_add_file.py \
-      --ignore=COT/commands/tests/test_edit_hardware.py \
-      --ignore=COT/commands/tests/test_edit_product.py \
-      --ignore=COT/commands/tests/test_edit_properties.py \
-      --ignore=COT/commands/tests/test_inject_config.py \
-      --ignore=COT/commands/tests/test_install_helpers.py \
-      --ignore=COT/commands/tests/test_remove_file.py \
-      --ignore=COT/disks/tests/test_disk_representation.py \
-      --ignore=COT/disks/tests/test_iso.py \
-      --ignore=COT/disks/tests/test_qcow2.py \
-      --ignore=COT/disks/tests/test_raw.py \
-      --ignore=COT/disks/tests/test_vmdk.py \
-      --ignore=COT/helpers/tests/test_helper.py \
-      --ignore=COT/helpers/tests/test_qemu_img.py \
-      --ignore=COT/tests/test_utilities.py \
-      --ignore=COT/ui/tests/test_cli.py \
-      --ignore=COT/vm_description/ovf/tests/test_item.py \
-      --ignore=COT/vm_description/ovf/tests/test_ovf.py
-    runHook postCheck
-  '';
-
+  checkInputs = [ mock pytest-mock pytestCheckHook ];
+  pytestFlagsArray = [
+     "--ignore=COT/commands/tests/test_add_disk.py"
+     "--ignore=COT/commands/tests/test_add_file.py"
+     "--ignore=COT/commands/tests/test_edit_hardware.py"
+     "--ignore=COT/commands/tests/test_edit_product.py"
+     "--ignore=COT/commands/tests/test_edit_properties.py"
+     "--ignore=COT/commands/tests/test_inject_config.py"
+     "--ignore=COT/commands/tests/test_install_helpers.py"
+     "--ignore=COT/commands/tests/test_remove_file.py"
+     "--ignore=COT/disks/tests/test_disk_representation.py"
+     "--ignore=COT/disks/tests/test_iso.py"
+     "--ignore=COT/disks/tests/test_qcow2.py"
+     "--ignore=COT/disks/tests/test_raw.py"
+     "--ignore=COT/disks/tests/test_vmdk.py"
+     "--ignore=COT/helpers/tests/test_helper.py"
+     "--ignore=COT/helpers/tests/test_qemu_img.py"
+     "--ignore=COT/tests/test_utilities.py"
+     "--ignore=COT/ui/tests/test_cli.py"
+     "--ignore=COT/vm_description/ovf/tests/test_item.py"
+     "--ignore=COT/vm_description/ovf/tests/test_ovf.py"
+    ];

   # argparse is part of the standardlib
   prePatch = ''
```

</details>

Reply:
disabledTests relate to filtering the test names. 

You should be able to use bash expansion as well, but I haven't used this with pytestFlagsArray:
```
"--ignore=COT/disks/tests/test_{vmdk,qcow2,raw,iso}.py"
```

Reply:
I'm also not a big fan of, "disable failing tests". Generally we should be only disabling tests that don't work well in a sandbox. Test failures maybe indicative that the package isn't fully usable.

Reply:
Yeah, many of the tests seem to use `ovftool`, and others seem to require network connectivity.
At some point, I would like to get `ovftool` added to nixpkgs, anyway, but for now, this is the list I have for `disabledTests`:
``` nix
  disabledTests = [
    "HelperGenericTest"
    "TestCOTAddDisk"
    "TestCOTAddFile"
    "TestCOTEditHardware"
    "TestCOTEditProduct"
    "TestCOTEditProperties"
    "TestCOTInjectConfig"
    "TestISO"
    "TestOVFAPI"
    "TestQCOW2"
    "TestRAW"
    "TestVMDKConversion"
  ];
```

```
=============== 367 passed, 210 deselected, 3 warnings in 5.61s ================
```

Reply:
Do you mind add that as a comment? :)